### ISSUE TITLE
Close Editor Keybinding on pinned tab should not move to next editor

### ISSUE THREAD

**Issue by mcandre** [microsoft/vscode#228778]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.93.1 (Universal)
- OS Version: macOS 14.6.1 (23G93)

I remember permatabs in Firefox and desire to get VSCode's pinned tabs operating more like that, which I find very intuitive. VSCode's pinned tab system is unfortunately somewhat awkward.

First, the pinned tab button is clickable. It resides in the exact same place that the close tab button normally resides. If I click on it twice, then the pinned tab closed. So I want an option to disable the click event altogether on the pin icon. I want to restrict VSCode to unpin only via the chord hotkey or the tab context menu. I cannot figure out how to accomplish this from the VSCode API documentation.

For now, I am hiding the close button and using the ordinary close tab hotkey as the primary way of closing my (unpinned) tabs. That way, I am less tempted to click on the pin/close button area.

Second, when I press the ordinary hotkey to close the hotkey  (`Command+W`  in macOS, `Control+W` elsewhere), then I expect absolutely nothing to happen. But if there are other non-pinned tabs open, then the navigation changes to the most recently used non-pinned tab. Not only is the event being mishandled in terms of taking an action, but it's not even respecting a semi-related preference to close and navigate left to right, instead of by recency. So at the end of the day, I don't want the active tab changed at all when using the hotkey on a pinned tab.

I tried some different when condition snippets for the second problem, but they either had no effect on application behavior, or prevented non-pinned tabs from closing.

I also tried showing an event log of stock VSCode button click events, in order to gather more information and build towards a settings.json configuration that would address these UX problems. But none of the Google results show how to debug VSCode's own DOM tree, only user project DOM trees.

Finally, there is a subtle width margin or width padding difference between pinned vs non-pinned tabs. This is revealed by quickly toggling between the two states, when the unpin button is hidden. I have not activated the setting to compact pinned tabs into just icons. I have activated the setting to apply fixed tab widths regardless of filename. So I wouldn't expect a tab width difference based on the pinned vs unpinned state alone.

**TL;DR. I would be satisfied with the addition of one more boolean options, to control whether the pin button is clickable, and a patch to the close tab handler so that it behaves as NO-OP whenever the tab is pinned.**

**Comment by benibenj** [comment]
@bpasero What is the reason for moving to the next tab when using the close editor keybinding? 
https://github.com/microsoft/vscode/blob/d66eb6b82f4fd73db17a4b7cc0799f90a09207bc/src/vs/workbench/browser/parts/editor/editorCommands.ts#L706

@mcandre we are not planning on making the pin/unpin action unclickable. You are able to hide it with `workbench.editor.tabActionUnpinVisibility` if that helps. It is also possible to move the action to the left of the tab with `workbench.editor.tabActionLocation`

**Comment by bpasero** [comment]
> @bpasero What is the reason for moving to the next tab when using the close editor keybinding?

So that you can quickly close all tabs when pressing the keybinding multiple times and not get stuck on a pinned one.

**Comment by benibenj** [comment]
To get the desired behaviour with `Ctrl + W` you can set the when clause to `!activeEditorIsPinned`.

Closing as we already have issues open regarding the other issues or are not planning to make any changes in the short term
